src/include/scriptingmanager.h

```diff
@@ -221,13 +264,52 @@ class DLLIMPORT ScriptingManager : public Mgr<ScriptingManager>, public wxEvtHan
         	cbThrow(_T("Can't assign a ScriptingManager* !!!"));
         	return *this;
 		}
+
+		ScriptBindings::CBSquirrelThread * CreateSandbox();
+
+		ScriptBindings::CBsquirrelVM* GetVM()       {return m_vm;};
+
+
+        /** \brief Search the plugin _Name_ and run its _Execute_ function
+         *
+         * \param Name wxString
+         * \return int  1 On success
+         *             -1 On script error. The error was reported by the script
+         *             -2 "Execute" could not be found in the script
+         *             -3 The script _Name_ could not be found
+         *
+         */
+		int ExecutePlugin(wxString Name);
+
+        /** \brief Search and return the Plugin with the _Name_
+         *
+         * \param Name wxString
+         * \return cbScriptPlugin* nullptr it not found
+         *
+         */
+		cbScriptPlugin* GetPlugin(wxString Name);
+
+        /** \brief Ask all script plugins to create the "right click" menu
+         *
+         * \param type const ModuleType
+         * \param menu wxMenu*
+         * \param data const FileTreeData*
+         * \return void
+         *
+         */
+		void CreateModuleMenu(const ModuleType type, wxMenu* menu, const FileTreeData* data);
+
+
     private:
         // needed for SqPlus bindings
         ScriptingManager(cb_unused const ScriptingManager& rhs); // prevent copy construction
 
         void OnScriptMenu(wxCommandEvent& event);
         void OnScriptPluginMenu(wxCommandEvent& event);
         void RegisterScriptFunctions();
+        void OnDebugTimer(wxTimerEvent& event);
```

Comment:
This has is without functionality at the moment and can be removed...
        void OnDebugTimer(wxTimerEvent& event);
        wxTimer m_DebugerUpdateTimer;